struct.c

```diff
@@ -806,6 +806,18 @@ struct_heap_alloc(VALUE st, size_t len)
     return ALLOC_N(VALUE, len);
 }
 
+bool
+rb_yarv_struct_instances_embedded_p(VALUE klass)
```

Comment:
I'm not sure this is the best way to do this. YJIT does it by inspecting an instance, which we don't have. Maybe we could add it to profiles? :/

Reply:
I am adding a new bit